### PR TITLE
Various cleanups in instrumentation-jvm

### PR DESCRIPTION
**Pull request by mimaison**

### CODE REVIEW

File: prometheus-metrics-instrumentation-jvm/src/test/java/io/prometheus/metrics/instrumentation/jvm/ProcessMetricsTest.java [hunk 1]
```diff
@@ -62,22 +62,32 @@ public void testGoodCase() throws IOException {
             + "process_max_fds 244.0\n"
             + "# TYPE process_open_fds gauge\n"
             + "# HELP process_open_fds Number of open file descriptors.\n"
-            + "process_open_fds 127.0\n"
-            + "# TYPE process_resident_memory_bytes gauge\n"
-            + "# UNIT process_resident_memory_bytes bytes\n"
-            + "# HELP process_resident_memory_bytes Resident memory size in bytes.\n"
-            + "process_resident_memory_bytes 1036288.0\n"
+            + "process_open_fds 127.0\n";
+    if (ProcessMetrics.PROC_SELF_STATUS.canRead()) {
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply prometheus-metrics-instrumentation-jvm/src/test/java/io/prometheus/metrics/instrumentation/jvm/ProcessMetricsTest.java:66]: can you add that as a comment it the code?

[the remark this answers is not in the document]
Comment (mimaison) [reply prometheus-metrics-instrumentation-jvm/src/test/java/io/prometheus/metrics/instrumentation/jvm/ProcessMetricsTest.java:67]: Yes, done

**Review by zeitlinger — APPROVED** [review]
great work :smile: 

if you want to continue cleaning up tests, I'd love to see assertj :100:

**Comment by mimaison** [comment]
There's an existing PR to upgrade to JUnit 5: https://github.com/prometheus/client_java/pull/908
I feel like JUnit 5 makes most of the common assertion patterns very simple so not sure if we really need assertj.

**Pull request merged**